### DISCUSSION TITLE
Use the library locally without pip!

### DISCUSSION THREAD

**Discussion by ghost** [rkhamilton/vqgan-clip-generator#52] [Q&A] [answered]
Hello!
First of all thanks for your work, amazing structure. 

I was wondering if there is a straightforward way to use your library locally instead of installing with pip3 on a conda environment. Maybe this is a python question, but here it goes!

Thank you

**Comment by gateway** [comment]
Not sure why you wouldn't want to use pip?

**Reply by ghost** [reply]
Just solved with `sys.path.append('vqgan-clip-generator/src')`! Thank you

**Comment by rkhamilton** [comment, marked as answer]
If you want to do development and edit the library (knock yourself out!) it's easy to do. Run the setup scripts, but instead of using pip to install it, you will clone it locally (copies the library to your computer within your current working directory), and then run setup in develop mode. Then it's available to use anywhere, but you have the files in front of you. And in develop mode, any changes you make in front of you are reflected throughout your system. This is how I work on my own PC as I develop the library. I just wanted an installable package for people who want to just make art, and not tinker.

To do this, run the setup script, except for the pip installing the library. Replace that with this:
```sh
git clone https://github.com/rkhamilton/vqgan-clip-generator.git
cd .\vqgan-clip-generator\
python setup.py develop
```

edit: If you want to avoid using conda and virtual environments, you can certainly just install everything normally.  I haven't tested any non-conda installers for compatibility though, so I can't give you a tested command for that.

**Reply by ghost** [reply]
Thank you very much @rkhamilton. Looking forward to more doubts and tinkering :)
